src/game/savegame/savegame_bson.c

```diff
@@ -575,18 +575,20 @@ static bool SaveGame_BSON_LoadFx(struct json_array_s *fx_arr)
         int16_t counter = json_object_get_int(fx_obj, "counter", counter);
         int16_t shade = json_object_get_int(fx_obj, "shade", shade);
 
-        int16_t fx_num = Effect_Create(room_num);
-        if (fx_num != NO_ITEM) {
-            FX_INFO *fx = &g_Effects[fx_num];
-            fx->pos.x = x;
-            fx->pos.y = y;
-            fx->pos.z = z;
-            fx->object_number = object_number;
-            fx->speed = speed;
-            fx->fall_speed = fall_speed;
-            fx->frame_number = frame_number;
-            fx->counter = counter;
-            fx->shade = shade;
+        if (g_Config.enable_enhanced_saves) {
+            int16_t fx_num = Effect_Create(room_num);
+            if (fx_num != NO_ITEM) {
+                FX_INFO *fx = &g_Effects[fx_num];
+                fx->pos.x = x;
+                fx->pos.y = y;
+                fx->pos.z = z;
+                fx->object_number = object_number;
+                fx->speed = speed;
+                fx->fall_speed = fall_speed;
+                fx->frame_number = frame_number;
+                fx->counter = counter;
+                fx->shade = shade;
+            }
```

Comment:
Shouldn't this entire function exit early if the setting is disabled?

Reply:
My idea was to save the FX regardless if the user enabled it in the JSON in case they ever want to turn it on. I wasn't sure if I could return early from this function because I don't completely understand how the savegame load works. So, I read all the FX data to be safe but just don't create the effect. I can change it if that's not the case.

Reply:
Yeah that's not necessary unless for validation sake.